client/src/components/assignment/CreateAssignment.js

```diff
@@ -0,0 +1,361 @@
+import Typography from '@material-ui/core/Typography';
+import Paper from '@material-ui/core/Paper';
+import Checkbox from '@material-ui/core/Checkbox';
+import { Link } from 'react-router-dom';
+const CreateAssignment = ({ getUsers, setSelectedUsers, user: { users, loading } }) => {
+
+    useEffect(() => {
+        getUsers();
+    }, [getUsers]);
+    const classes = useStyles();
+    const [order, setOrder] = React.useState('asc');
+    const [orderBy, setOrderBy] = React.useState('calories');
+    const [idsSelected, setIdsSelected] = React.useState([]);
+    const [selected, setSelected] = React.useState([]);
+    const [page, setPage] = React.useState(0);
+    const [rowsPerPage, setRowsPerPage] = React.useState(5);
+
+    const handleRequestSort = (event, property) => {
+        const isDesc = orderBy === property && order === 'desc';
+        setOrder(isDesc ? 'asc' : 'desc');
+        setOrderBy(property);
+    };
+
+    const handleSelectAllClick = event => {
+        if (event.target.checked) {
+            const newSelecteds = users.map(n => n.name);
+            const selected_ids = users.map(n => n._id);
+            setSelected(newSelecteds);
+            setIdsSelected(selected_ids);
+            return;
+        }
+        setSelected([]);
+        setIdsSelected([]);
+    };
+
+    const handleClick = (event, id, name) => {
+        const selectedIndex = idsSelected.indexOf(id);
+        let newSelected = [];
+        let newIdsSelected = [];
+        if (selectedIndex === -1) {
+            newSelected = newSelected.concat(selected, name);
+            newIdsSelected = newIdsSelected.concat(idsSelected, id);
+        } else if (selectedIndex === 0) {
+            newSelected = newSelected.concat(selected.slice(1));
+            newIdsSelected = newIdsSelected.concat(idsSelected.slice(1));
+        } else if (selectedIndex === selected.length - 1) {
+            newSelected = newSelected.concat(selected.slice(0, -1));
+            newIdsSelected = newIdsSelected.concat(idsSelected.slice(0, -1));
+        } else if (selectedIndex > 0) {
+            newSelected = newSelected.concat(
+                selected.slice(0, selectedIndex),
+                selected.slice(selectedIndex + 1),
+            );
+            newIdsSelected = newIdsSelected.concat(
+                idsSelected.slice(0, selectedIndex),
+                idsSelected.slice(selectedIndex + 1),
+            );
+        }
+        setSelected(newSelected);
+        setIdsSelected(newIdsSelected);
+    };
+
+    const handleChangePage = (event, newPage) => {
+        setPage(newPage);
+    };
+
+    const handleChangeRowsPerPage = event => {
+        setRowsPerPage(+event.target.value);
+        setPage(0);
+    };
+
+    const isSelected = name => selected.indexOf(name) !== -1;
+
+    const emptyRows = rowsPerPage - Math.min(rowsPerPage, users.length - page * rowsPerPage);
+
+    const onNextButtonClick = e => setSelectedUsers(idsSelected, selected);
```

Comment:
This might be in the css, but I think when the button is greyed out, the cursor shouldn't be a pointing hand, but the prohibited circle. That or add an alert if they try to click it without selecting anything, that tells them to select at least one user before clicking next.

Reply:
Fixed!